public/app/core/components/AppChrome/TopBar/TopNavBarMenu.tsx

```diff
@@ -1,18 +1,25 @@
 import { css } from '@emotion/css';
 import { i18n } from '@lingui/core';
+import { cloneDeep } from 'lodash';
 import React from 'react';
+import { useLocation } from 'react-router-dom';
 
 import { GrafanaTheme2, NavModelItem } from '@grafana/data';
 import { Menu, MenuItem, useStyles2 } from '@grafana/ui';
 
 import menuItemTranslations from '../../NavBar/navBarItem-translations';
+import { enrichConfigItems, enrichWithInteractionTracking } from '../../NavBar/utils';
 
 export interface TopNavBarMenuProps {
   node: NavModelItem;
 }
 
-export function TopNavBarMenu({ node }: TopNavBarMenuProps) {
+export function TopNavBarMenu({ node: nodePlain }: TopNavBarMenuProps) {
   const styles = useStyles2(getStyles);
+  const location = useLocation();
+  const enriched = enrichConfigItems([cloneDeep(nodePlain)], location);
+  const node = enrichWithInteractionTracking(enriched[0], false);
```

Comment:
now useLocation, enrich + tracking is only done for a single tree node. And this component is only rendered when the Dropdown overlay is opened